pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -0,0 +1,402 @@
+	} else if err := pc.syncPod(key); err == nil {
+		pc.queue.Forget(key)
+	} else {
+		pc.queue.AddRateLimited(key)
+	}
+	return true
+}
+
+// Configure SRIOV VF as a Secondary Network Interface.
+func (pc *PodController) configureSriovAsSecondaryInterface(pod *corev1.Pod, netinfo *SecondaryNetworkObject, containerInfo *cnipodcache.CNIConfigInfo, result *current.Result) error {
+	podSriovVFDeviceID, err := cniserver.GetPodContainerDeviceIDs(pod.Name, pod.Namespace)
+	if err != nil {
+		return fmt.Errorf("GetPodContainerDeviceIDs failed: %v", err)
+	}
+
+	if err = pc.cniServer.GetPodConfigurator().ConfigureSriovSecondaryInterface(
+		containerInfo.PodName,
+		containerInfo.PodNameSpace,
+		containerInfo.ContainerID,
+		containerInfo.ContainerNetNS,
+		netinfo.InterfaceName,
+		containerInfo.MTU,
+		podSriovVFDeviceID[0],
+		result,
+	); err != nil {
+		return fmt.Errorf("SRIOV Interface creation failed: %v", err)
+	}
+	return nil
+}
+
+// Configure Secondary Network Interface.
+func (pc *PodController) configureSecondaryInterface(pod *corev1.Pod, netinfo *SecondaryNetworkObject, podCNIInfo []*cnipodcache.CNIConfigInfo, cniconfig []byte) error {
+	var err error
+	var ipamerr error
+	var ipamResult *current.Result
+	var cmdArgs *invoke.Args
+
+	// Assigne default interface name, If secondary interface name was not provided in Pod annotation.
+	if len(netinfo.InterfaceName) == 0 {
+		netinfo.InterfaceName = defaultSecondaryInterfaceName
+	}
+
+	for _, containerInfo := range podCNIInfo {
+		// CNI Server CmdDel handle will update CNI cache entry's PodStatus as "podDelComplete" upon successful clean-up of primary network config.
+		// Here (configureSecondaryInterface) we check for the PodStatus entry and do not process cache (as part of the current Pod Add event handle sequence), if its
+		// already marked as podDelComplete by CNI Server. CNI cache with PodStatus set to "podDelComplete" will be cleaned up as part of handleRemovePod (Pod Del event handle).
+		// NOTE: This additional status check was handled to avoid duplicate cache processing for the same PodName and PodNamespace.
+		// PodSecondaryIface shows the status of secondary network interface configuration. If already configured, avoid processing it again (As per the current implementation,
+		// we do not have full support for Pod update event handle (i'e, Pod annotation changes etc.,).
+		// TBD: Impelment Linux interface configuration API to look for secondary network interface in container scope and replace PodSecondaryIface check with the new API (Ideal
+		// way to handle).
+		if containerInfo.PodCNIStatus != podDelComplete && containerInfo.PodSecondaryIface == false {
+			// Update Virtual Network configuration (retrieved from network-attach-definition) at CNI cache. This is required to release IP address at whereabouts DB.
+			containerInfo.NetworkInfo = cniconfig
+			containerInfo.ContainerIfDev = netinfo.InterfaceName
+			pc.podCache.UpdateCNIConfigInfo(containerInfo)
```

Comment:
I feel this is not needed, as containerInfo is a pointer and we already updated the fields. But could you very it?